src/smallmap_gui.cpp

```diff
@@ -1004,6 +1009,45 @@ class SmallMapWindow : public Window {
 		}
 	}
 
+	/**
+	 * Adds industry names to the smallmap.
+	 * @param dpi the part of the smallmap to be drawn into
+	 */
+	void DrawIndustryNames(const DrawPixelInfo *dpi) const
+	{
+		if (this->map_type != SMT_INDUSTRY) return;
+
+		for (const Industry *i : Industry::Iterate()) {
+			const LegendAndColour &tbl = _legend_from_industries[_industry_to_list_pos[i->type]];
+			if (!tbl.show_on_map) continue;
+
+			const bool is_blinking = i->type == _smallmap_industry_highlight && !_smallmap_industry_highlight_state;
```

Comment:
Might be nice to have a comment here, `/* Industry names blink with their blob on the smallmap. */` or similar.